klippy/extras/probe.py

```diff
@@ -224,9 +243,10 @@ def move_next(self):
         self.gcode.reset_last_position()
     cmd_NEXT_help = "Move to the next XY position to probe"
     def cmd_NEXT(self, params):
-        # Record current position
-        self.toolhead.wait_moves()
-        self.results.append(self.callback.get_probed_position())
+        if not self.probe:
```

Comment:
Is this change necessary?  (It's not a problem to check for self.probe, but wouldn't NEXT only ever be called when self.probe is None.)

Separately, as a random side-note, it may be safer in python to explicitly check for `if some_class is None:` then to check for truth (`if some_class:`), as some classes may end up implementing the truth callback.

Reply:
Hi Kevin, unless I am missing something, self.cmd_NEXT({}) is called on line 249 in the busy loop.  If I don't make the check then an extra point will be appended after the probe_point() function completes.

I did make the suggested change to the truth check.